lib/rdoc/yard_plugin.rb

```diff
@@ -0,0 +1,228 @@
+          if current_indentation_level >= @base_indentation_level + 2
+            # Append to the previous tag
+            data = @mode == :param ? @parsed_comment[@mode].last : @parsed_comment[@mode]
+            data.append_desc(line)
+          else
+            if (tag, matchdata = matching_any_tag(line))
+              if tag == :param
+                type = matchdata[:type1] || matchdata[:type2]
+                name = matchdata[:name1] || matchdata[:name2]
+                desc = matchdata[:desc1] || matchdata[:desc2]
+                parsed_type = TypeParser.parse(type)
+                @parsed_comment[:param] << ParamData.new(type: parsed_type, name: name, desc: desc)
+                @mode = :param
+              elsif tag == :return
+                type = matchdata[:type]
+                desc = matchdata[:desc]
+                parsed_type = TypeParser.parse(type)
+                @parsed_comment[:return] = ReturnData.new(type: parsed_type, desc: desc)
+                @mode = :return
+              elsif tag == :raise
+                type = matchdata[:type]
+                desc = matchdata[:desc]
+                parsed_type = TypeParser.parse(type)
+                @parsed_comment[:raise] << RaiseData.new(type: parsed_type, desc: desc)
+                @mode = :raise
+              end
+            else
+              @parsed_comment[:plain] << line
+            end
+          end
+          @base_indentation_level = current_indentation_level
+        end
+
+        @parsed_comment
+      end
+
+      private
+
+      def matching_any_tag(line)
+        TAG_PARSING_REGEXES.each do |tag, regex|
+          matchdata = line.match(regex)
+          return [tag, matchdata] if matchdata
+        end
+        nil
+      end
+    end
+
+    class Type
+      attr_reader :name
+
+      def initialize(name)
+        @name = name
+      end
+
+      def to_s
+        @name
+      end
+    end
+
+    class CollectionType < Type
+      attr_reader :type
+
+      def initialize(name, type)
+        super(name)
+        @type = type
+      end
+
+      def to_s
+        "#{@name}<#{@type}>"
+      end
+    end
+
+    class FixedCollectionType < Type
+      attr_reader :type
+
+      def initialize(name, type)
+        super(name)
+        @type = type
+      end
+
+      def to_s
+        "#{@name}(#{@type})"
+      end
+    end
+
+    class HashCollectionType < Type
+      attr_reader :key_type, :value_type
+
+      def initialize(name, key_type, value_type)
+        super(name)
+        @key_type = key_type
+        @value_type = value_type
+      end
+
+      def to_s
+        "#{@name}<#{@key_type} => #{@value_type}>"
+      end
+    end
+
+    class TypeParser
+      TOKENS = {
+        collection_start: /</,
+        collection_end: />/,
+        fixed_collection_start: /\(/,
+        fixed_collection_end: /\)/,
+        type_name: /#\w+|((::)?\w+)+/,
+        literal: /(?:
+           '(?:\\'|[^'])*' |
+           "(?:\\"|[^"])*" |
```

Comment:
I think some regexps are wrong.
`"\"` (escaped, quote is not closed) matches to this string literal regexp.
Integer and Float literal(integer part) matches to `type_name: /#\w+|((::)?\w+)+/`

Reply:
Oh I didn't notice this. I tested this code with some variations of type and it seems working. Tests are missing and we need to add them to ensure this regex working correctly since it's quite complex.